.github/workflows/unit-tests.yml

```diff
@@ -1,51 +1,84 @@
+          sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
+
+      - name: Install Minikube
+        run: |
+          curl -Lo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64
+          chmod +x minikube
+          sudo mv minikube /usr/local/bin/
+
+      - name: Set up Minikube
+        env:
+          GOOGLE_APPLICATION_CREDENTIALS: /tmp/google_application_credentials.json
+        run: |
+          echo "${{ secrets.TEST_GCP_SERVICE_KEY }}" | base64 --decode > /tmp/google_application_credentials.json
+          bash bin/test-env-up.sh
+
+          # Manually add the minikube host alias because it sometimes doens't work and google won't fix it
+          # https://github.com/kubernetes/minikube/issues/8439
+          eval $(minikube docker-env --unset)
+          sudo chown $USER /etc/hosts
+          echo $(minikube ssh grep host.minikube.internal /etc/hosts | cut -f1) host.minikube.internal >> /etc/hosts
+          eval $(minikube docker-env)
+
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
 
       - name: Install dependencies
         run: |
+          sudo apt-get update
           sudo apt-get install pigz -y
-          
           python -m pip install --upgrade pip
 
           cd ..
-          git clone https://github.com/The-Academic-Observatory/observatory-platform.git
-          cd observatory-platform
-          pip install -e observatory-api --constraint https://raw.githubusercontent.com/apache/airflow/constraints-2.7.3/constraints-no-providers-${{ matrix.python-version }}.txt
-          pip install -e observatory-platform --constraint https://raw.githubusercontent.com/apache/airflow/constraints-2.7.3/constraints-no-providers-${{ matrix.python-version }}.txt
-          cd ..
+          # TODO: Change back!!
+          # git clone https://github.com/The-Academic-Observatory/observatory-platform.git
+          git clone -b feature/astro-refactor https://github.com/The-Academic-Observatory/observatory-platform.git
+          pip install -e ./observatory-platform[tests] --constraint https://raw.githubusercontent.com/apache/airflow/constraints-2.7.3/constraints-no-providers-${{ matrix.python-version }}.txt
 
           cd academic-observatory-workflows
-          pip install -e .[tests] --constraint https://raw.githubusercontent.com/apache/airflow/constraints-2.7.3/constraints-no-providers-${{ matrix.python-version }}.txt
+          pip install -e ./academic-observatory-workflows[test] --constraint https://raw.githubusercontent.com/apache/airflow/constraints-2.7.3/constraints-no-providers-${{ matrix.python-version }}.txt
 
           cd ..
           curl -LO https://github.com/peak/s5cmd/releases/download/v2.1.0/s5cmd_2.1.0_linux_amd64.deb
           sudo dpkg -i s5cmd_2.1.0_linux_amd64.deb
 
+          pip freeze
+          airflow db init
```

Comment:
Why do we need to init the Airflow DB? Or is this because we are using the Airflow's dag.test?

Reply:
Yeah this was an issue I was having. I couldn't interact with the database at all and it didn't seem to exist so I added this line and it was happy. No idea why it isn't necessary when running locally but is required here.